public/partials/farm_new.html

```diff
@@ -50,7 +51,9 @@
     </md-input-container>
 
     <md-input-container layout="row" layout-align="center center">
-        <md-button ng-click="cancelAdd()">Cancel</md-button>
-        <md-button ng-click="addNewFarm()">Save</md-button>
+        <md-button ng-if="newFarm" ng-click="cancelAdd()">Cancel</md-button>
```

Comment:
It seems almost identical to what I've already done though. Similar use of ng-ifs. What's the main difference that you would like me to implement?